File: CSharp/SolutionLib/Public.cs
using System;
using System.Collections.Generic;

namespace SolutionLib
{
    public class Public
    {
        public static string ReadStr()
        {
            Console.WriteLine("输入一个字符串：");
            return ReadLine();
        }

        public static string[] ReadStrs()
        {
            Console.WriteLine("输入一组字符串，以回车结尾：");
            return GetStrings(ReadLine());
        }

        public static int ReadNum()
        {
            Console.WriteLine("输入一个数，以回车结尾：");
            return GetNum(ReadLine());
        }

        public static int[] ReadNums()
        {
            Console.WriteLine("输入一组数，以回车结尾：");
            return GetNums(ReadLine());
        }

        public static int[][] ReadMatrix()
        {
            Console.WriteLine("输入一个二维数组，以回车结尾：");
            return Get2DNums(ReadLine());
        }

        public static ListNode ReadList(int[] nums)
        {
            if (nums.Length == 0) return null;
            var head = new ListNode(nums[0]);
            var pt = head;
            for (int i = 1; i < nums.Length; i++)
            {
                pt.next = new ListNode(nums[i]);
                pt = pt.next;
            }
            return head;
        }

        public static ListNode ReadList()
        {
            return ReadList(ReadNums());
        }

        public static void Print<T>(IEnumerable<T> results)
        {
            bool flag = false;
            Console.Write('[');
            foreach (var result in results)
            {
                if (flag) Console.Write(",");
                Console.Write(result);
                flag = true;
            }
            Console.Write(']');

            Console.WriteLine();
        }

        public static void Print(ListNode head, bool isHead)
        {
            if (head == null)
            {
                Console.Write(isHead ? "链表为空\n" : "");
                return;
            }
            Console.Write((isHead ? "" : "->") + head.val);
            Print(head.next, false);

            if (isHead)
                Console.WriteLine();
        }

        private static string ReadLine()
        {
            return Console.ReadLine();
        }

        private static int GetNum(string str)
        {
            return int.Parse(str);
        }

        private static string[] GetStrings(string str)
        {
            if (str.Length <= 2) return null;
            var res = new List<string>();
            str = str.Substring(1, str.Length - 2);
            foreach (var s in str.Split(','))
            {
                if (s == "") continue;
                res.Add(s);
            }

            return res.ToArray();
        }

        private static int[] GetNums(string str)
        {
            var res = new List<int>();
            if (str.Length <= 2)
                return res.ToArray();
            foreach (var num in GetStrings(str))
                res.Add(GetNum(num));

            return res.ToArray();
        }

        private static int[][] Get2DNums(string str)
        {
            if (str.Length <= 4) return new int[][] { };
            var res = new List<int[]>();
            str = str.Substring(1, str.Length - 2);
            str = str.Replace("],[", "].[");
            foreach (var nums in str.Split('.'))
            {
                res.Add(GetNums(nums));
            }

            return res.ToArray();
        }

    }
}

File: Java/src/UtilLib/Public.java
package UtilLib;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Public {
    private static BufferedReader scanner = new BufferedReader(new InputStreamReader(System.in));

    public static String readStr() {
        System.out.println("输入一行字符串，以回车结尾：");
        String str = readLine();
        if (str.charAt(0) == '"') str = str.substring(1, str.length() - 1);
        return str;
    }

    public static int readNum() {
        System.out.println("输入一个数，以回车结尾：");
        return Integer.parseInt(readLine());
    }

    public static int[] readNums() {
        System.out.println("输入一组数，以回车结尾：");
        return getNums(readLine());
    }

    private static String readLine() {
        try {
            return scanner.readLine();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return "";
    }

    private static int[] getNums(String str) {
        String readLine = str;
        if (readLine.charAt(0) == '[')
            readLine = readLine.substring(1, readLine.length() - 1);
        var split = readLine.split(",");
        var res = new int[split.length];
        for (int i = 0; i < split.length; i++) {
            res[i] = Integer.parseInt(split[i]);
        }
        return res;
    }

}
